File: rust/src/server/color.rs
pub const TRANSPARENT: i8 = -1;
pub const BLACK: i8 = 0;
pub const WHITE: i8 = 1;
pub const RED: i8 = 2;
pub const CYAN: i8 = 3;
pub const PURPLE: i8 = 4;
pub const GREEN: i8 = 5;
pub const BLUE: i8 = 6;
pub const YELLOW: i8 = 7;
pub const ORANGE: i8 = 8;
pub const BROWN: i8 = 9;
pub const LIGHT_RED: i8 = 10;
pub const DARK_GREY: i8 = 11;
pub const GREY: i8 = 12;
pub const LIGHT_GREEN: i8 = 13;
pub const LIGHT_BLUE: i8 = 14;
pub const LIGHT_GREY: i8 = 15;

pub fn get_palette() -> Vec<(i8, &'static str)> {
    vec![
        (BLACK, "#000000"),
        (WHITE, "#fcf9fc"),
        (RED, "#933a4c"),
        (CYAN, "#b6fafa"),
        (PURPLE, "#d27ded"),
        (GREEN, "#6acf6f"),
        (BLUE, "#4f44d8"),
        (YELLOW, "#fbfb8b"),
        (ORANGE, "#d89c5b"),
        (BROWN, "#7f5307"),
        (LIGHT_RED, "#ef839f"),
        (DARK_GREY, "#575753"),
        (GREY, "#a3a7a7"),
        (LIGHT_GREEN, "#b7fbbf"),
        (LIGHT_BLUE, "#a397ff"),
        (LIGHT_GREY, "#d0d0d0"),
    ]
}

File: ts/src/server/color.ts
export enum Color {
  BLACK = 0,
  WHITE = 1,
  RED = 2,
  CYAN = 3,
  PURPLE = 4,
  GREEN = 5,
  BLUE = 6,
  YELLOW = 7,
  ORANGE = 8,
  BROWN = 9,
  LIGHT_RED = 10,
  DARK_GREY = 11,
  GREY = 12,
  LIGHT_GREEN = 13,
  LIGHT_BLUE = 14,
  LIGHT_GREY = 15,
  TRANSPARENT = -1,
}

const palette: Record<number, string> = {
  [Color.BLACK]: "#000000",
  [Color.WHITE]: "#fcf9fc",
  [Color.RED]: "#933a4c",
  [Color.CYAN]: "#b6fafa",
  [Color.PURPLE]: "#d27ded",
  [Color.GREEN]: "#6acf6f",
  [Color.BLUE]: "#4f44d8",
  [Color.YELLOW]: "#fbfb8b",
  [Color.ORANGE]: "#d89c5b",
  [Color.BROWN]: "#7f5307",
  [Color.LIGHT_RED]: "#ef839f",
  [Color.DARK_GREY]: "#575753",
  [Color.GREY]: "#a3a7a7",
  [Color.LIGHT_GREEN]: "#b7fbbf",
  [Color.LIGHT_BLUE]: "#a397ff",
  [Color.LIGHT_GREY]: "#d0d0d0",
};

export function getPalette(): Readonly<Record<number, string>> {
  return palette;
}
